### ISSUE TITLE
Slot will not be filled - every time None

### ISSUE THREAD

**Issue by mrRobot62** [RasaHQ/rasa_core#771]
**Rasa Core version**:
$ >pip list | grep rasa
rasa-core                     0.9.8      
rasa-nlu                      0.12.3  

**Python version**: 
3.6.6

**Operating system** (windows, osx, ...):
mac os 
Darwin MacBook-Pro-4.local 17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64
**Issue**:

**Content of domain file** (if used & relevant):
```
intents:
- greet
- goodbye
- user_need_help
- inform

entities:
- user_need_help
- username

actions:
- utter_user_authorization
- utter_greet
- utter_goodbye
- coco.ActionHelp
- utter_greet_user

templates:
    utter_user_authorization:
        - "I have to authorize you. Please enter your first name"

    utter_greet:
        - "Nice to meet you. What can I do for you?"

    utter_goodbye:
        - "Hope, I could help you. Good bye"

    utter_greet_user:
        - "Hi {username}"
slots:
    username:
        type: text
```

as an example from data.json
...
      {
        "text": "my name is peter",
        "intent": "inform",
        "entities": [
          {
            "start": 11,
            "end": 16,
            "value": "peter",
            "entity": "username"
          }
        ]
      },
      {
        "text": "here is frank",
        "intent": "inform",
        "entities": [
          {
            "start": 8,
            "end": 13,
            "value": "frank",
            "entity": "username"
          }
        ]
      },
      {
        "text": "here is marie-luise",
        "intent": "inform",
        "entities": [
          {
            "start": 8,
            "end": 19,
            "value": "marie-luise",
            "entity": "username"
          }
        ]
      },
...

## Story 01a
* greet
    - utter_greet
    - utter_user_authorization
* inform
    - utter_greet_user
* goodbye
    - utter_goodbye

# story 02
* goodbye
  - utter_goodbye

** Problem: **

I trained NLU and rasa_core-dialog - no error messages
Than I started training_online

Example (during online training)
user: hi
Bot: utter_greet (ok)
Bot: utter_user_authorization (ok) => action_listen (ok)
user: my name is bernd
bot:
	user said:	my name is bernd

		 whose intent is:	inform

we currently have slots: username: None

------
The bot wants to [utter_greet_user] due to the intent. Is this correct?

Itent : OK
Action: ok
Seems entity will not be recognized
and is username not filled?
Entity (see above) is "username"

I believe it's a rookie error - but I worked a couple of hours on it :-( - hope somebody give me a kick :-)

Cheers

**Comment by akelad** [comment]
How much training data do you have for this entity? Names are quite hard to extract so you need quite a bit of training data

**Comment by mrRobot62** [comment]
at least 8 data entries in where I inserted "entity":"username".

Text examples
`"my name is peter",`
`"my name is hans-peter"`
`"hans-peter"`
`"here is frank"`
`"here is marie-luise"`
`"I'm Bernd"`
`"I'm peter-pan"`
`"myname"`

 All above are like this:
      {
        "text": "I'm peter-pan",
        "intent": "inform",
        "entities": [
          {
            "start": 4,
            "end": 13,
            "value": "peter-pan",
            "entity": "username"
          }
        ]
      },

**Comment by mrRobot62** [comment]
BTW: I create some more intents with entity "username". During train_nlu I got below warnings 

`UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/sklearn/metrics/classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
`

Can I ignore them?

**Comment by mrRobot62** [comment]
ok, but I'm (honestly) a little bit surprised, because, in your examples, with weather-bot and the entity location. It works, isn't it?
From a technical point of view, between "location" and "username" is not a big difference and 
second point - both are type: text -  If my bot ask for an name. I assume the user write most time only "his/her" name and maybe sometimes "My name...".  What kind of "training data should I include more?

Is the length of the "input string : like Frank or Bernhard or Bob" relevant? If yes, than I understood your recommendation, to include more training data . If not, I'm a little bit confused ;-)

**Comment by mrRobot62** [comment]
I inserted a couple of more entries for first names. But I believe there is something wrong in general.
As attachment I include my project. Hop I do not forget something.

Don't worry about the MOCK python file, it's for later use but it's called in the current script.

Hope that somebody can help me to fix my issue

Thank's a lot to the community to help me.

[myIssue.zip](https://github.com/RasaHQ/rasa_core/files/2216056/myIssue.zip)

**Comment by akelad** [comment]
The issue with names is there's no real pattern there, and so it's quite hard for NLU to pick up consistently. What you can definitely try as well is using `ner_spacy`, this has an inbuilt entity called `PERSON`, which should help pick up some of the names. 
As for your question about what other training data to add - more different kinds of names in the sentence structures you already have

**Comment by mrRobot62** [comment]
ok, will give `ner_spacy` a chance :-). Hope it will fix it.
But as I said, I do not 100% understand, what is the technical difference between 
"in what location do you want to eat?" and user insert " Berlin"
or
"Hi, what is you name?" and user insert "Bernd"
both are entities, which we can use and both should store the value in a slot with a type text. (e.g. location and username)


mmhm, I believe, I do not understand the context behind the scene :-O

**Comment by akelad** [comment]
What's obvious to a human sadly is not always obvious to a machine :D location names are actually also quite hard to recognise as soon as you start varying them

**Comment by akelad** [comment]
Is ner_spacy helping?

**Comment by mrRobot62** [comment]
not 100%, but with ner_spacy - better. Sometimes I got a value in a slot, sometimes not :-/
Changed to FormAction (with all handicaps behind) as well, that's much more better but I'm struggle around ;-)

Before I start on scratch for my "real" bot I will have a small simple bot which I can used with training_online, slot filling (context), form actions and a straightforward dialog flow

**Comment by akelad** [comment]
Yeah sadly person names are quite hard. I'm afraid there's no obvious solution there, apart from adding more training data with time and it will get better! 
I'll close this issue for now then, if you have any more issues feel free to open a new one
